File: FineApple/src/main/java/com/fineapple/domain/shipment/entity/Shipment.java
package com.fineapple.domain.shipment.entity;

import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@NoArgsConstructor
public class Shipment {
    private Long shipmentId;
    private String trackingNumber;
    private String trackingUrl;
    private LocalDateTime estimatedDeliveryDate;
    private LocalDateTime dispatchedAt; //출고일시
    private LocalDateTime deliveredAt; // 배송 완료 일시
    private LocalDateTime updatedAt;
    private String deliveryStatus;
    private Long storeId;
    private Long deliveryId;
    private BigDecimal shippingCost;
    private String courierCompany;
    private String delayReason;
    private String deliveryAgentName;
    private String deliveryAgentContact;
    private String postNum;
    private String city;
    private String region;
    private String roadNum;
    private String address;
    private String storeName;
    private Long orderId;

    @Builder
    public Shipment(Long shipmentId, String trackingNumber, String trackingUrl, LocalDateTime estimatedDeliveryDate,
                    LocalDateTime dispatchedAt, LocalDateTime deliveredAt, LocalDateTime updatedAt, String deliveryStatus,
                    Long storeId, Long deliveryId, BigDecimal shippingCost, String courierCompany, String delayReason,
                    String deliveryAgentName, String deliveryAgentContact, String postNum, String city, String region,
                    String roadNum, String address, String storeName, Long orderId) {
        this.shipmentId = shipmentId;
        this.trackingNumber = trackingNumber;
        this.trackingUrl = trackingUrl;
        this.estimatedDeliveryDate = estimatedDeliveryDate;
        this.dispatchedAt = dispatchedAt;
        this.deliveredAt = deliveredAt;
        this.updatedAt = updatedAt;
        this.deliveryStatus = deliveryStatus;
        this.storeId = storeId;
        this.deliveryId = deliveryId;
        this.shippingCost = shippingCost;
        this.courierCompany = courierCompany;
        this.delayReason = delayReason;
        this.deliveryAgentName = deliveryAgentName;
        this.deliveryAgentContact = deliveryAgentContact;
        this.postNum = postNum;
        this.city = city;
        this.region = region;
        this.roadNum = roadNum;
        this.address = address;
        this.storeName = storeName;
        this.orderId = orderId;
    }
}

File: FineApple/src/main/resources/static/js/shipment.js
// document.addEventListener('DOMContentLoaded', () => {
//     // shipment 페이지일 경우에만 실행
//     if (window.location.pathname === '/shipment') {
//         fetchShipments();
//     }
// });
//
// async function fetchShipments() {
//     try {
//         const res = await fetch('/api/v1/shipments');
//         if (!res.ok) throw new Error(`HTTP ${res.status}`);
//         const shipmentList = await res.json();
//         renderShipmentTable(shipmentList);
//     } catch (err) {
//         console.error('🚨 배송 정보 불러오기 실패:', err);
//     }
// }
//
// function renderShipmentTable(data) {
//     const tbody = document.getElementById('shipmentTableBody');
//     tbody.innerHTML = ''; // 초기화
//
//     if (!data.length) {
//         tbody.innerHTML = `<tr><td colspan="10" style="text-align:center;">배송 정보가 없습니다.</td></tr>`;
//         return;
//     }
//
//     data.forEach(s => {
//         const row = `
//             <tr>
//                 <td>${s.shipmentId}</td>
//                 <td>${s.trackingNumber}</td>
//                 <td><a href="${s.trackingUrl}" target="_blank">${s.trackingUrl}</a></td>
//                 <td>${formatDate(s.estimatedDeliveryDate)}</td>
//                 <td>${formatDate(s.dispatchedAt)}</td>
//                 <td>${formatDate(s.deliveredAt)}</td>
//                 <td>${s.shippingCost}</td>
//                 <td>${s.deliveryAgentName ?? '(없음)'}</td>
//                 <td>${s.city} ${s.region} ${s.roadNum} ${s.address}</td>
//                 <td>${s.storeName}</td>
//             </tr>
//         `;
//         tbody.insertAdjacentHTML('beforeend', row);
//     });
// }
//
// function formatDate(dateString) {
//     if (!dateString) return '-';
//     return new Date(dateString).toLocaleString(); // yyyy-MM-dd HH:mm
// }
//
//
// async function loadShipmentPage() {
//     try {
//         const res = await fetch('/shipment');
//         if (!res.ok) throw new Error(`HTML 페이지 로딩 실패: ${res.status}`);
//         const html = await res.text();
//         const parser = new DOMParser();
//         const doc = parser.parseFromString(html, 'text/html');
//         const newMain = doc.querySelector('main');
//         const currentMain = document.querySelector('main');
//
//         if (newMain && currentMain) {
//             currentMain.replaceWith(newMain);
//             fetchShipments(); // 새 main 영역에 shipment 데이터 렌더링
//         }
//     } catch (error) {
//         console.error('🚨 배송 페이지 불러오기 실패:', error);
//     }
// }
//
// // 배송 아이콘 클릭 시 동적 로딩 연결
// document.querySelector('.local-mall')?.addEventListener('click', (e) => {
//     e.preventDefault();
//     loadShipmentPage();
// });